### ISSUE TITLE
Bun doesn't work on Windows 10

### ISSUE THREAD

**Issue by Kolobamanacas** [uptrace/bun#83]
Below is simple example of code, which falls with "Authentication error".

* The DB server is up and running and testDatabase exists.
* postgres user exists and all access rights to the testDatabase are granted.
* I intentionally put the simplest password possible to avoid typos.
* I'm quite sure that credentials are correct, cause changing host, port, login or password to wrong values leads to another appropriate errors, like "Host doesn't exist" or "Wrong login/password".
* Connection to the same database with the same credentials works fine both from SQL-client (Azure Data Studio) and another app (using NodeJS/TypeORM).

So whether I'm doing something wrong, or this is a bug indeed.

Here the code:

```go
package main

import (
	"context"
	"database/sql"

	"github.com/gin-gonic/gin"
	"github.com/uptrace/bun/driver/pgdriver"
	"github.com/uptrace/bun/extra/bundebug"

	"github.com/uptrace/bun"
	"github.com/uptrace/bun/dialect/sqlitedialect"
)

func main() {
	r := gin.Default()

	conn := sql.OpenDB(pgdriver.NewConnector(pgdriver.WithDSN("postgres://postgres:1@localhost:5432/testDatabase?sslmode=disable")))
	db := bun.NewDB(conn, sqlitedialect.New())
	db.AddQueryHook(bundebug.NewQueryHook(bundebug.WithVerbose()))

	ctx := context.Background()

	_, err := db.ExecContext(ctx, "SELECT 1")
	if err != nil {
		panic(err)
	}

	if err := db.Ping(); err != nil {
		panic(err)
	}

	r.GET("/ping", func(c *gin.Context) {
		c.JSON(200, gin.H{
			"message": "pong",
		})
	})
	r.Run(":8080") // listen and serve on 0.0.0.0:8080
}
```

And here is the error:

```
> go run .\main.go
[GIN-debug] [WARNING] Creating an Engine instance with the Logger and Recovery middleware already attached.

[GIN-debug] [WARNING] Running in "debug" mode. Switch to "release" mode in production.
 - using env:   export GIN_MODE=release
 - using code:  gin.SetMode(gin.ReleaseMode)

[bun]  10:27:03.447   SELECT               34.707ms  SELECT 1     *errors.errorString: Authentication error
panic: Authentication error

goroutine 1 [running]:
main.main()
        C:/Users/kolob/Desktop/buntest/main.go:26 +0x457
exit status 2
```

**Comment by vmihailenco** [comment]
You should replace `sqlitedialect` with `pgdialect`, but I am not sure it will fix the problem.

You can check if `sql.DB` is working first:

```go
sqldb := sql.OpenDB(pgdriver.NewConnector(pgdriver.WithDSN("postgres://postgres:1@localhost:5432/testDatabase?sslmode=disable")))
fmt.Println(sqldb.Ping())
```

If the problem persists, try pgx:

```go
config, err := pgx.ParseConfig("postgres://postgres:1@localhost:5432/testDatabase?sslmode=disable")
if err != nil {
	panic(err)
}

config.PreferSimpleProtocol = true
sqldb := stdlib.OpenDB(*config)
```

If the problem persists with pgx, then your connect credentials are wrong...

**Comment by Kolobamanacas** [comment]
Thanks a lot for your reply.

Replacing `sqlitedialect` with `pgdialect` didn't help. But with `pgx`, the second option you mentioned, it works like a charm. It seems like not a credentials issue after all. Below is the code I used.

```go
package main

import (
	"context"
	"fmt"
	"os"

	_ "github.com/go-pg/pg/v10/orm"
	"github.com/jackc/pgx/v4"
)

func main() {
	conn, err := pgx.Connect(context.Background(), "postgres://postgres:1@localhost:5432/testDatabase?sslmode=disable")
	if err != nil {
		fmt.Fprintf(os.Stderr, "Unable to connect to database: %v\n", err)
		os.Exit(1)
	}
	defer conn.Close(context.Background())

	var id int
	err = conn.QueryRow(context.Background(), "select 1").Scan(&id)

	if err != nil {
		fmt.Fprintf(os.Stderr, "QueryRow failed: %v\n", err)
		os.Exit(1)
	}

	fmt.Println(id)
}
```

**Comment by Perplex** [comment]
Same issue, sql.OpenDB also returns an authentication error.

**Comment by vmihailenco** [comment]
Thanks for the report - it was a bug with SCRAM-256 authentication. It should be fixed in v0.3.7
